### ISSUE TITLE
PDF as image

### ISSUE THREAD

**Issue by prowlett** [coursebuilder-ncl/makecourse#4]
In LaTeX I can use `\includegraphics{}` to use a PDF as an image. I tried using a PDF as an image with `makecourse` and it appeared in the PDF document but not the web version. Does it take PDF as an image file, or do I need to convert the PDF to something web-friendly before running?

**Comment by christianp** [comment]
It looks like plasTeX won't convert PDFs automatically. It should fall back to using `latex` to render images, but it doesn't seem to add the source path to `TEXINPUTS`.

We should either intercept pdfs in our plasTeX Renderer, or find a way of fixing the `TEXINPUTS` environment variable so plasTeX's call to latex will find the input pdf.

For now, convert PDFs to SVGs. `pdf2svg` is a good command-line tool available on ubuntu.

**Comment by prowlett** [comment]
The use of PDF as image is likely to be to include some diagram produced from LaTeX using a package not supported here, like a `tikz` diagram. A workaround would be to make the diagram in a `standalone` doc as a PDF file, then use `\includegraphics` to include that in the notes. 

> For now, convert PDFs to SVGs.

Pending `makecourse` dealing with the PDF included via `\includegraphics`, a workaround is to make the figure document using `\documentclass[convert]{standalone}` and then compile it using `pdflatex -shell-escape file.tex` to create a PNG file directly (from [an answer on StackExchange](https://tex.stackexchange.com/a/11880)), which is then accepted by plasTeX as an image file in `\includegraphics`.

**Comment by prowlett** [comment]
Hmm, `\documentclass[convert=pdf2svg]{standalone}` creates an SVG alright, but then `makecourse` reports an error with `pdflatex` which boils down to `! LaTeX Error: Unknown graphics extension: .svg.` 

I originally had the comment as using `\documentclass[convert]{standalone}` to create a PNG, which could be included with `\includegraphics` in the usual way, but edited this to the SVG version. 

Because you said 

> For now, convert PDFs to SVGs.

@christianp, can you say how to load SVG graphics into documents in a way that is compatible with `makecourse`?

**Comment by extigy** [comment]
I believe the problems using PDF images with `\includegraphics{}` are largely caused by using the `dvipng` non-vector imager in plasTeX, which I have explicitly configured makecourse to use. The imager tries to build images with `latex` in DVI mode and it seems that when `latex` is in this mode it does not support reading the size of PDF images unless the `dvipdfm` option is used when loading `graphicx`.

So, I have created a branch (dc396c70530f91f042c3cbfef9a602f08f736e17) where the default imager is instead set to `pdftoppm`. This imager produces images using `pdflatex` with PDF as the intermediate format so should hopefully support loading PDF files with `\includegraphics{}` much better. I've also added a sample PDF and tikz figure to a similar branch in the sample course (coursebuilder-ncl/makecourse@1fee08877a0444106d20d33f37cbb5bb087b52db) as a way to test the vector and non-vector imagers.

I've held off pushing straight to master since this requires an additional prerequisite (`poppler-utils` on Ubuntu). I'll need to test things in a few places I'm running makecourse (including in docker) and make sure our automated module build system still works okay before merging.
